auacm/app/modules/submission_manager/views.py

```diff
@@ -69,3 +69,44 @@ def submit():
     return serve_response({
         'submissionId': attempt.job
     })
+
+
+@app.route('/api/submit/job/<int:job_id>')
+def get_submit(job_id):
+    """Get the current status of a submission based on its id"""
+    submit = (session.query(models.Submission)
+              .filter(models.Submission.job == job_id).first())
+
+    if not submit:
+        return serve_error('Submission with job id ' + str(job_id) +
+                           ' not found', 401)
+
+    return serve_response(submit.to_dict())
+
+
+@app.route('/api/submit/user/<username>')
+@app.route('/api/submit/user/<username>/<int:limit>')
```

Comment:
This was way harder to figure out than I thought it would be.

Can we change these to something more apt? I understand if we want to do something like see recent submissions for a user accessible to anyone, but I think we should rename these. Making the route default to the current user would make sense; that's what we'll do most of the time anyway. The use of a query string here would make a lot of sense; it can serve as a sort of filter. Here is what I propose:

For all users: `/api/submissions`

For getting another user's submissions (accessible through `request.args.get('username')`): `/api/submissions?username=<username>`

For getting submissions with a limit (accessible through `request.args.get('limit')`): `/api/submissions?limit=<int:limit>`

For the currently logged-in user (I kind of hate this one, but this is as good as I could come up with): `/api/submissions/mine`

While we're making these changes, I think we should also make a change to the judge route. I wrote it before I really knew what I was doing. The route for it should likely be something like `/api/submissions` but with the method `POST`. I can do this one in a new branch if you'd like.